uaclient/cli/formatter.py

```diff
@@ -57,3 +69,14 @@ def decolorize(self):
     def split(self, /, sep=None, maxsplit=-1):
         splitted = super().split(sep=sep, maxsplit=maxsplit)
         return [self.__class__(s) for s in splitted]
+
+
+def colorize(input_string: str, color: TxtColor) -> str:
+    if not ProOutputFormatterConfig.use_color:
+        return input_string
+
+    return_string = ProColorString(color.value + input_string)
+    if not return_string.endswith(TxtColor.ENDC.value):
+        return_string += TxtColor.ENDC.value
```

Comment:
This logic: not adding the color codes:
because if no-color is set, then there is no need we return a ProColorString at all. Each `len` call (although inexpensive) involves a quick regex check there. The logic could be moved there for sure, I just thought cutting it in the roots would make it easier for all function calls after this is executed.
(the `len` is implemented there to adjust the wording to screens, wrapping table columns, etc; so it will be called several times per run)

This logic: not adding an extra endc:
- this definitely is meant to be here. It could be removed though - it dos not *hurt* to have double ENDCs but I found it annoying

Reply:
My main rationale here is that the ProColorString should know how to "build" itself. My rationale is that we should only create a `ProColorString(color=color, text=text)` and it should return the formatted string already

Reply:
Oki I tried it, and it limits us a bit. What happens is that the `ProColorString` instance is in fact a string, and needs to behave as a string. This means :
- it may have no color at all (and that's fine, it just means there is nothing to decolorize).
- it may have the color just in some slice of the string (it contains the color control characters in the middle of the string, not necessarily in the start/end)

This behavior gives us the flexibility to concatenate those instances together, and also to concatenate them with regular strings and keep the `len` behavior - which is what we want here in the end.

I'd keep it as is.